### PR TITLE
feat(pydantic): use pydantic serialization

### PR DESCRIPTION
**Pull request by MartinBelthle**
fix [ANT-2052]

### CODE REVIEW

File: antarest/utils.py [hunk 1]
```diff
@@ -9,265 +9,30 @@
...
-
-    services["event_bus"] = event_bus
-    services["study"] = study_service
-    services["launcher"] = launcher
-    services["matrix"] = matrix_service
-    services["user"] = user_service
-    services["cache"] = cache
-    services["maintenance"] = maintenance_service
-    return services
+    # indent: int | None = None,
```
Comment (MartinBelthle) [inline antarest/utils.py:28]: Remove this

File: tests/cache/test_redis_cache.py [hunk 2]
```diff
@@ -10,12 +10,12 @@
 #
 # This file is part of the Antares project.
 
-import json
 from pathlib import Path
 from unittest.mock import Mock
 
 from antarest.core.cache.business.redis_cache import RedisCache, RedisCacheElement
 from antarest.study.storage.rawstudy.model.filesystem.config.model import Area, FileStudyTreeConfigDTO
+from antarest.utils import from_json
```
Comment (MartinBelthle) [inline tests/cache/test_redis_cache.py:18]: check isort and black

File: antarest/tools/lib.py [hunk 3]
```diff
@@ -207,10 +208,7 @@ def extract_commands(study_path: Path, commands_output_dir: Path) -> None:
     command_list = extractor.extract(study)
 
     (commands_output_dir / COMMAND_FILE).write_text(
-        json.dumps(
-            [command.model_dump(exclude={"id"}) for command in command_list],
-            indent=2,
-        )
+        to_json([command.model_dump(exclude={"id"}) for command in command_list], indent=2).decode("utf-8")
```
Comment (sylvlecl) [inline antarest/tools/lib.py]: we can provide 2 methods, one to get bytes, the other to get strings, to not have to repeat the `decode` everywhere we need a string

File: antarest/utils.py [hunk 4]
```diff
@@ -9,265 +9,18 @@
...
-
-    services["event_bus"] = event_bus
-    services["study"] = study_service
-    services["launcher"] = launcher
-    services["matrix"] = matrix_service
-    services["user"] = user_service
-    services["cache"] = cache
-    services["maintenance"] = maintenance_service
-    return services
+def to_json(data: t.Any, indent: t.Optional[int] = None) -> bytes:
```
Comment (sylvlecl) [inline antarest/utils.py]: What would you think about moving this to a `core.serialization` module ? Something more targetted than the plain "utils".

Also we should add a docstring to mention that we do this to benefit from the performances of pydantic serialization

File: antarest/core/serialization/utils.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+def from_json(data: t.Union[str, bytes, bytearray]) -> t.Dict[str, t.Any]:
+    return ADAPTER.validate_json(data)  # type: ignore
+
+
+def to_json(data: t.Any, indent: t.Optional[int] = None) -> bytes:
+    return ADAPTER.dump_json(data, indent=indent)
+
+
+def to_json_as_bytes(data: t.Any, indent: t.Optional[int] = None) -> str:
```
Comment (sylvlecl) [inline antarest/core/serialization/utils.py]: `to_json_string` would be more appropriate :)

File: antarest/eventbus/web.py [hunk 6]
```diff
@@ -99,7 +99,7 @@ async def send_event_to_ws(event: Event) -> None:
         event_data = event.model_dump()
         del event_data["permissions"]
         del event_data["channel"]
-        await manager.broadcast(json.dumps(event_data), event.permissions, event.channel)
+        await manager.broadcast(to_json(event_data).decode("utf-8"), event.permissions, event.channel)
```
Comment (sylvlecl) [inline antarest/eventbus/web.py]: we can use the new method to get the string here

File: antarest/core/configdata/repository.py [hunk 7]
```diff
@@ -43,14 +43,14 @@ def get(self, key: str, owner: Optional[int] = None) -> Optional[ConfigData]:
...
-            data: JSON = json.loads(configdata.value)
+            data: JSON = from_json(configdata.value)
             return data
         return None
 
     def put_json(self, key: str, data: JSON, owner: Optional[int] = None) -> None:
         configdata = ConfigData(
             key=key,
-            value=json.dumps(data),
+            value=to_json(data).decode("utf-8"),
```
Comment (sylvlecl) [inline antarest/core/configdata/repository.py]: we can use the new method

File: antarest/study/storage/variantstudy/variant_study_service.py [hunk 8]
```diff
@@ -188,7 +188,10 @@ def append_commands(
         # noinspection PyArgumentList
         new_commands = [
             CommandBlock(
-                command=command.action, args=json.dumps(command.args), index=(first_index + i), version=command.version
+                command=command.action,
+                args=to_json(command.args).decode("utf-8"),
```
Comment (sylvlecl) [inline antarest/study/storage/variantstudy/variant_study_service.py]: new method should be used

File: antarest/study/storage/variantstudy/variant_study_service.py [hunk 9]
```diff
@@ -314,7 +319,7 @@ def update_command(
         index = [command.id for command in study.commands].index(command_id)
         if index >= 0:
             study.commands[index].command = validated_commands[0].action
-            study.commands[index].args = json.dumps(validated_commands[0].args)
+            study.commands[index].args = to_json(validated_commands[0].args).decode("utf-8")
```
Comment (sylvlecl) [inline antarest/study/storage/variantstudy/variant_study_service.py]: new method should be used

File: antarest/study/storage/variantstudy/variant_study_service.py [hunk 10]
```diff
@@ -223,7 +226,9 @@ def replace_commands(
         validated_commands = transform_command_to_dto(command_objs, commands)
         # noinspection PyArgumentList
         study.commands = [
-            CommandBlock(command=command.action, args=json.dumps(command.args), index=i, version=command.version)
+            CommandBlock(
+                command=command.action, args=to_json(command.args).decode("utf-8"), index=i, version=command.version
```
Comment (sylvlecl) [inline antarest/study/storage/variantstudy/variant_study_service.py]: new method should be used

File: antarest/core/configdata/repository.py [hunk 11]
```diff
@@ -10,13 +10,13 @@
...
 # This file is part of the Antares project.
 
-import json
 from operator import and_
 from typing import Optional
 
 from antarest.core.configdata.model import ConfigData
 from antarest.core.jwt import DEFAULT_ADMIN_USER
 from antarest.core.model import JSON
+from antarest.core.serialization.utils import from_json, to_json
```
Comment (sylvlecl) [inline antarest/core/configdata/repository.py]: I was more thinking of removing the `utils` part (just antarest.core.serialization)

**Review by sylvlecl — COMMENTED** [review]
the use of the new method to get a string should be generalized

**Comment by MartinBelthle** [comment]
Indeed I was lazy

**Review by sylvlecl — APPROVED** [review]

**Pull request merged**